src/plugins/ecs/ecs_compares.py

```diff
@@ -0,0 +1,305 @@
+			def_name_list.remove(environment_code_name)
+			if len(def_name_list) > 2:
+				result.append(def_name_list[0] + "_" + service_name)
+				result.append(service_name)
+			else:
+				result.append(service_name)
+
+	return result
+
+def build_compare_words(lookup, compareto, jenkin_build_terms):
+
+	result = False
+
+	if compareto:
+		if "-" in compareto:
+			compareto = compareto.replace("-", "_")
+		if " " in compareto:
+			compareto = compareto.replace(" ", "_")
+		compareto = compareto.lower().split("_")
+	else:
+		compareto = []
+
+	if lookup:
+		if "-" in lookup:
+			lookup = lookup.replace("-", "_")
+		if " " in lookup:
+			lookup = lookup.replace(" ", "_")
+		lookup = lookup.lower().split("_")
+	else:
+		lookup = []
+
+	#aggregate unique values in the two lists
+	res = list(set(compareto) ^ set(lookup))
+
+	if len(res) == 2 and jenkin_build_terms[0] in res and jenkin_build_terms[2] in res:
+		result = True
+	elif len(res) == 1 and (jenkin_build_terms[0] in res or jenkin_build_terms[1] in res):
+		result = True
+
+	return result
+
+def get_build_url(cached_array, lookup_word, prelim_version, jenkins_tags):
+
+	the_url = None
+
+	#compare the service name to links in the superjenkins_data
+	# set the build_url when a url contains words matching the lookup service name
+	for the_names in cached_array:
+		if build_compare_words(lookup_word, the_names['name'], jenkins_tags):
+			the_url = the_names['url']
+
+	symbols_array = [".", "_", "-"]
+
+	build_num = []
+
+	#extract the build number from version
+	for symb in symbols_array:
+		if symb in prelim_version:
+			build_num = prelim_version.split(symb)
+			break
+
+	if len(build_num) > 1 and the_url:
+		final_url = str(the_url) + build_num[-1] + "/promotion/ | ver: " + str(prelim_version)
+		final_url = "<" + final_url + ">"
+	else:
+		# build url corresponding to service was not found
+		final_url = "ver: " + str(prelim_version)
+
+	return final_url
+
+	# compare master to teams
+def ecs_compare_master_team(tkey, m_array, cached_array, jenkins_build_tags, excluded_services=None):
+
+	compared_array = {}
+	ecs_data = []
+
+	for eachmaster in m_array:
+		for m_data in m_array[eachmaster]:
+
+
+			for t_array in tkey:
+				if t_array['servicename'] == m_data['servicename']:
+
+					logging.debug("Printing comparison of service_definition")
+					logging.debug(t_array['service_definition'] + " == " + m_data['service_definition'])
+
+					# check if service_name is on excluded services list
+					do_not_exclude_service = True
+					for ex_service in excluded_services:
+						if ex_service in t_array['servicename']:
+							do_not_exclude_service = False
+
+					if do_not_exclude_service:
+
+						the_team_service_name = finalize_service_name(t_array['servicename'],
+																	  t_array['service_definition'],
+																	  t_array['environment_code_name'])
+
+						the_master_service_name = finalize_service_name(m_data['servicename'],
+																		m_data['service_definition'],
```

Comment:
What is up with the huge white space at the beginning of this line, and the next - also down below...